larq_compute_engine/core/bitpack.h

```diff
@@ -33,10 +33,80 @@ inline void bitpack_bitfield_quantized(const TIn* in, TBitpacked* out,
                                        const TIn zero_point) {
   ruy::profiler::ScopeLabel label(
       "Bitpack bitfield (quantized input, unoptimised)");
-  *out = 0;
-  for (size_t i = 0; i < bitpacking_bitwidth; ++i) {
-    if (in[i] < zero_point) *out |= (TBitpacked(1) << i);
-  }
+  struct bf {
+    unsigned int b0 : 1;
+    unsigned int b1 : 1;
+    unsigned int b2 : 1;
+    unsigned int b3 : 1;
+    unsigned int b4 : 1;
+    unsigned int b5 : 1;
+    unsigned int b6 : 1;
+    unsigned int b7 : 1;
+    unsigned int b8 : 1;
+    unsigned int b9 : 1;
+    unsigned int b10 : 1;
+    unsigned int b11 : 1;
+    unsigned int b12 : 1;
+    unsigned int b13 : 1;
+    unsigned int b14 : 1;
+    unsigned int b15 : 1;
+    unsigned int b16 : 1;
+    unsigned int b17 : 1;
+    unsigned int b18 : 1;
+    unsigned int b19 : 1;
+    unsigned int b20 : 1;
+    unsigned int b21 : 1;
+    unsigned int b22 : 1;
+    unsigned int b23 : 1;
+    unsigned int b24 : 1;
+    unsigned int b25 : 1;
+    unsigned int b26 : 1;
+    unsigned int b27 : 1;
+    unsigned int b28 : 1;
+    unsigned int b29 : 1;
+    unsigned int b30 : 1;
+    unsigned int b31 : 1;
+  };
+
+  union bf_i32 {
+    bf t;
+    TBitpacked i32;
+  };
+
+  bf_i32 i;
+  i.t.b0 = in[0] < zero_point;
```

Comment:
Out of interest, did you try benchmarking with `i.t.b0 = (int[0] - zero_point) >> 7`?

Reply:
Or I guess this is actually generic over TIn, so using numeric-limits to get num digits.

Reply:
Good idea, I did not try. Will do now!

Reply:
That turned out to be roughly 5% slower.